checker/check-components-schemas-removed.go

```diff
@@ -0,0 +1,28 @@
+package checker
+
+import (
+	"github.com/tufin/oasdiff/diff"
+)
+
+const (
+	apiSchemasRemovedCheckId = "api-model-removed"
+)
+
+func APIOperationSchemasRemovedCheck(diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, config BackwardCompatibilityCheckConfig) []BackwardCompatibilityError {
+	result := make([]BackwardCompatibilityError, 0)
+	if diffReport.ComponentsDiff.SchemasDiff == nil {
+		return result
+	}
+
+	for _, pathItem := range diffReport.ComponentsDiff.SchemasDiff.Deleted {
+		result = append(result, BackwardCompatibilityError{
+			Id:        apiSchemasRemovedCheckId,
+			Level:     ERR,
+			Text:      config.i18n(apiSchemasRemovedCheckId),
+			Operation: "",
```

Comment:
It is tricky to figure out the operation and source for the model as we never know the source of deletion.
Deletion can happen because

- the model was never used in paths (and we cannot ensure that is the case)
- Model was renamed

because oasdiff doesn't return diff for the paths when only ref was renamed (I guess we cannot change that)
we cannot build operation name

Reply:
Hi @wtrocki ,
I compared the two specs below and got a diff.
Does this correspond with your requirement?

Comparing the two specs below gives:
```
paths:
    modified:
        /partner-api/test/some-method:
            ref:
                from: '#/components/paths/path1'
                to: '#/components/paths/path2'
```

base spec:
```
openapi: 3.0.1
info:
  title: Test API
  version: v1
paths:
  /partner-api/test/some-method:
    $ref: '#/components/paths/path1'
components:
  paths:
    path1:
      get:
        responses:
          "200":
            description: Success
```

revision spec:
```
openapi: 3.0.1
info:
  title: Test API
  version: v1
paths:
  /partner-api/test/some-method:
    $ref: '#/components/paths/path2'
components:
  paths:
    path2:
      get:
        responses:
          "200":
            description: Success
```

Reply:
I might have used old version then. Then I believe that we can pick up ref changes and make this rule much better.
Thank you for reply.

Reply:
I see it now. It looks like only paths would show as diff but not actual refs to other elements like schemas or parameters.

My schema:
```
openapi: 3.0.1
info:
  title: Test API
  version: v1
paths:
  /partner-api/test/some-method:
    get:
      responses:
        200:
          description: OK
      requestBody:
        content:
          application/json:
            schema:     
              $ref: '#/components/schemas/path2'
components:
  schemas:
      path2:
        properties:
          name:
            type: string
```
base.yaml
```
openapi: 3.0.1
info:
  title: Test API
  version: v1
paths:
  /partner-api/test/some-method:
    get:
      responses:
        200:
          description: OK
      requestBody:
        content:
          application/json:
            schema:     
              $ref: '#/components/schemas/path1'
components:
  schemas:
      path1:
        properties:
          name:
            type: string
```

 ./oasdiff --base base.yaml --revision revison.yaml
components:
    schemas:
        added:
            - path2
        deleted:
            - path1

// Missing info about change in endpoint.

Reply:
This is related to the fact that loading specs resolve references instantly. 
That means that we only look into the reference itself but never have the ability to check for the reference name to be different. 

If we were to make this change in diff.go that would likely be breaking change?

Reply:
Seems like you found a bug:
References under paths are compared only by ref name - oasdiff should compare the ref values. 

In addition, if I understand correctly, you would like oasdiff to be able to compare $ref names, especially for paths but potentially for other elements.

Please let me know if I understood correctly.

Reply:
Yes. I'm trying to actually make fix by creating `RefDiff` object right now.
If any type of $ref have been changed in the OpenAPI it show come as diff. 
For this specific case we looking for components schemas.

Reply:
To find use case outside openapi generated clients:
If someone changes the schema and it is lucky enough to point to model with the same property (name is the same but it might have different/breaking meaning (description)).

Reply:
I will try to propose changes in diff section in this PR.
If you prefer separate one let me know.

Reply:
See https://github.com/Tufin/oasdiff/issues/276